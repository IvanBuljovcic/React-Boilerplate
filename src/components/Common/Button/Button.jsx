import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ text, type, onClick }) => {
  switch (type) {
    case 'reset':
      return (
        <input
          type="reset"
          className="app-btn app-btn--cancel"
          value="Cancel"
          onClick={onClick}
        />
      );

    case 'submit':
      return (
        <button type="submit" className="app-btn app-btn--submit" onClick={onClick}>
          Submit
        </button>
      );

    default:
      return (
        <button className="app-btn" type="button" onClick={onClick}>
          {text}
        </button>
      );
  }
};

Button.propTypes = {
  text: PropTypes.string,
  type: PropTypes.oneOf(['reset', 'submit', 'button'])
};

Button.defaultProps = {
  text: 'Submit',
  type: 'button'
};

export default Button;
