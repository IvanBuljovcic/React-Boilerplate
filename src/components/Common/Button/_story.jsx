import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs/react';
import StoryRouter from 'storybook-react-router';

// - Component
import Button from './Button';

const stories = storiesOf('Common', module);

stories.addDecorator(withKnobs);
stories.addDecorator(StoryRouter());

stories.addWithChapters('Buttons', {
  chapters: [
    {
      sections: [
        {
          title: 'Default',
          info: 'Default button component. Accepts a text prop.',
          sectionFn: () => <Button />,
        },
        {
          title: 'Submit',
          info: 'Submit button, does not accept the text prop',
          sectionFn: () => <Button type="submit" />,
        },
        {
          title: 'Reset',
          info: 'Reset button, does not accept the text prop',
          sectionFn: () => <Button type="reset" />,
        },
      ],
    },
  ],
});
