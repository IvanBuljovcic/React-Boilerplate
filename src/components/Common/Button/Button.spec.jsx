import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('<Button /> component renders correctly', () => {
  test('Default button', () => {
    const tree = renderer.create(<Button />).toJSON();

    expect(tree).toMatchSnapshot();
  })

  test('Submit button', () => {
    const tree = renderer.create(<Button type="submit" />).toJSON();

    expect(tree).toMatchSnapshot();
  })

  test('Reset button', () => {
    const tree = renderer.create(<Button type="reset" />).toJSON();

    expect(tree).toMatchSnapshot();
  })
})

