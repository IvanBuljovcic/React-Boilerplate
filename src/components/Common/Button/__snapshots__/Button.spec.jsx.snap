// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> component renders correctly Default button 1`] = `
<button
  className="app-btn"
  type="button"
>
  I'm a button
</button>
`;

exports[`<Button /> component renders correctly Reset button 1`] = `
<input
  className="app-btn app-btn--cancel"
  type="reset"
  value="Cancel"
/>
`;

exports[`<Button /> component renders correctly Submit button 1`] = `
<button
  className="app-btn app-btn--submit"
  type="submit"
>
  Submit
</button>
`;
